static/app/views/issueDetails/streamline/issueCronCheckTimeline.tsx

```diff
@@ -0,0 +1,243 @@
+import {Fragment, useMemo, useRef} from 'react';
+import styled from '@emotion/styled';
+
+import {CheckInPlaceholder} from 'sentry/components/checkInTimeline/checkInPlaceholder';
+import {CheckInTimeline} from 'sentry/components/checkInTimeline/checkInTimeline';
+import {
+  Gridline,
+  GridLineLabels,
+  GridLineOverlay,
+} from 'sentry/components/checkInTimeline/gridLines';
+import {useTimeWindowConfig} from 'sentry/components/checkInTimeline/hooks/useTimeWindowConfig';
+import type {StatsBucket} from 'sentry/components/checkInTimeline/types';
+import {Flex} from 'sentry/components/container/flex';
+import {Tooltip} from 'sentry/components/tooltip';
+import {tct} from 'sentry/locale';
+import {space} from 'sentry/styles/space';
+import type {Event} from 'sentry/types/event';
+import type {Group} from 'sentry/types/group';
+import {useApiQuery} from 'sentry/utils/queryClient';
+import {useDebouncedValue} from 'sentry/utils/useDebouncedValue';
+import {useDimensions} from 'sentry/utils/useDimensions';
+import useOrganization from 'sentry/utils/useOrganization';
+import {useUser} from 'sentry/utils/useUser';
+import {useIssueDetails} from 'sentry/views/issueDetails/streamline/context';
+import {getGroupEventQueryKey} from 'sentry/views/issueDetails/utils';
+import {MonitorIndicator} from 'sentry/views/monitors/components/monitorIndicator';
+import {CheckInStatus, type MonitorBucket} from 'sentry/views/monitors/types';
+import {
+  checkInStatusPrecedent,
+  statusToText,
+  tickStyle,
+} from 'sentry/views/monitors/utils';
+import {selectCheckInData} from 'sentry/views/monitors/utils/selectCheckInData';
+import {useMonitorStats} from 'sentry/views/monitors/utils/useMonitorStats';
+
+export function useCronIssueAlertId({groupId}: {groupId: string}): string | undefined {
+  /**
+   * This should be removed once the cron rule value is set on the issue.
+   * This will fetch an event from the max range if the detector details
+   * are not available (e.g. time range has changed and page refreshed)
+   */
+  const user = useUser();
+  const organization = useOrganization();
+  const {detectorDetails} = useIssueDetails();
+  const {detectorId, detectorType} = detectorDetails;
+
+  const hasCronDetector = detectorId && detectorType === 'cron_monitor';
+
+  const {data: event} = useApiQuery<Event>(
+    getGroupEventQueryKey({
+      orgSlug: organization.slug,
+      groupId,
+      eventId: user.options.defaultIssueEvent,
+      environments: [],
+    }),
+    {
+      staleTime: Infinity,
+      enabled: !hasCronDetector,
+      retry: false,
+    }
+  );
+
+  // Fall back to the fetched event since the legacy UI isn't nested within the provider the provider
+  return hasCronDetector
+    ? detectorId
+    : event?.tags?.find(({key}) => key === 'monitor.id')?.value;
+}
+
+function useCronLegendStatuses({bucketStats}: {bucketStats: MonitorBucket[]}) {
```

Comment:
adding a type here might help later
```suggestion
function useCronLegendStatuses({bucketStats}: {bucketStats: MonitorBucket[]}): CheckInStatus[] {
```